From aad45d5cdc81d35a5645c52eb964506a79052a0e Mon Sep 17 00:00:00 2001
From: pixelos-gsi <pixelos-gsi@localhost>
Date: Fri, 9 Oct 2026 19:48:49 +0000
Subject: [PATCH 11/11] partial:
 0001-settings-drop-the-vm-capability-check-from-the-lin.patch

Change-Id: I44797e277b24eecc3d7ef54decb2152a6ce1bead
---
 .../LinuxTerminalPreferenceController.java       |  7 +------
 tests/robotests/Android.bp                       |  1 -
 .../LinuxTerminalPreferenceControllerTest.java   | 16 ----------------
 3 files changed, 1 insertion(+), 23 deletions(-)

diff --git a/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceController.java b/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceController.java
index 93d3a0b..66f5cff 100644
--- a/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceController.java
+++ b/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceController.java
@@ -58,16 +58,11 @@ public class LinuxTerminalPreferenceController extends DeveloperOptionsPreferenc
                 isPackageInstalled(context.getPackageManager(), packageName) ? packageName : null;
 
         StorageManager storageManager = context.getSystemService(StorageManager.class);
-        VirtualMachineManager virtualMachineManager =
-                context.getSystemService(VirtualMachineManager.class);
 
         mIsDeviceCapable =
                 getTotalMemory() >= MEMORY_MIN_BYTES
                         && storageManager != null
-                        && storageManager.getPrimaryStorageSize() >= STORAGE_MIN_BYTES
-                        && virtualMachineManager != null
-                        && ((virtualMachineManager.getCapabilities() & CAPABILITY_NON_PROTECTED_VM)
-                                != 0);
+                        && storageManager.getPrimaryStorageSize() >= STORAGE_MIN_BYTES;
     }
 
     // Avoid lazy initialization because this may be called before displayPreference().
diff --git a/tests/robotests/Android.bp b/tests/robotests/Android.bp
index 4ebee9e..b7d3df6 100644
--- a/tests/robotests/Android.bp
+++ b/tests/robotests/Android.bp
@@ -91,7 +91,6 @@ android_robolectric_test {
 
     libs: [
         "android.test.mock.impl",
-        "framework-virtualization.stubs.system",
         "ims-common",
         "telephony-common",
     ],
diff --git a/tests/robotests/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceControllerTest.java b/tests/robotests/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceControllerTest.java
index d17b49b..8001c07 100644
--- a/tests/robotests/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceControllerTest.java
+++ b/tests/robotests/src/com/android/settings/development/linuxterminal/LinuxTerminalPreferenceControllerTest.java
@@ -16,7 +16,6 @@
 
 package com.android.settings.development.linuxterminal;
 
-import static android.system.virtualmachine.VirtualMachineManager.CAPABILITY_NON_PROTECTED_VM;
 
 import static com.android.settings.development.linuxterminal.LinuxTerminalPreferenceController.MEMORY_MIN_BYTES;
 import static com.android.settings.development.linuxterminal.LinuxTerminalPreferenceController.STORAGE_MIN_BYTES;
@@ -34,7 +33,6 @@ import android.content.pm.PackageInfo;
 import android.content.pm.PackageManager;
 import android.content.pm.PackageManager.NameNotFoundException;
 import android.os.storage.StorageManager;
-import android.system.virtualmachine.VirtualMachineManager;
 
 import org.junit.Before;
 import org.junit.Test;
@@ -51,7 +49,6 @@ public class LinuxTerminalPreferenceControllerTest {
     @Mock private Context mContext;
     @Mock private PackageManager mPackageManager;
     @Mock private StorageManager mStorageManager;
-    @Mock private VirtualMachineManager mVirtualMachineManager;
     @Mock private PackageInfo mPackageInfo;
 
     private LinuxTerminalPreferenceController mController;
@@ -69,10 +66,6 @@ public class LinuxTerminalPreferenceControllerTest {
         doReturn(mStorageManager).when(mContext).getSystemService(StorageManager.class);
         doReturn(STORAGE_MIN_BYTES).when(mStorageManager).getPrimaryStorageSize();
 
-        doReturn(mVirtualMachineManager)
-                .when(mContext)
-                .getSystemService(VirtualMachineManager.class);
-        doReturn(CAPABILITY_NON_PROTECTED_VM).when(mVirtualMachineManager).getCapabilities();
     }
 
     @Test
@@ -91,15 +84,6 @@ public class LinuxTerminalPreferenceControllerTest {
         assertThat(mController.isAvailable()).isFalse();
     }
 
-    @Test
-    public void isAvailable_whenVmNotSupported_returnFalse() {
-        doReturn(0).when(mVirtualMachineManager).getCapabilities();
-
-        mController = createController(mContext);
-
-        assertThat(mController.isAvailable()).isFalse();
-    }
-
     @Test
     public void isAvailable_whenPackageExists_returnsTrue() {
         mController = createController(mContext);
-- 
2.34.1

